parquet-column/src/main/java/org/apache/parquet/io/RecordReaderImplementation.java

```diff
@@ -424,13 +434,105 @@ public T read() {
     return record;
   }
 
+  private State getState(MessageType schema) {
+    for (State state : states) {
+        if (schema.containsField(state.fieldPath[0])) {
+          return state;
+        }
+    }
+    throw new IllegalArgumentException("Couldn't find state for schema " + schema);
+  }
+
   @Override
-  public boolean shouldSkipCurrentRecord() {
-    return shouldSkipCurrentRecord;
+  public void readVectors(ColumnVector[] vectors, MessageType[] columnSchemas, long current, long total) {
+      for (int i = 0 ; i < columnSchemas.length; i++) {
+        MessageType schema  = columnSchemas[i];
+        State state = getState(schema);
+        readVector(state.column, vectors[i], current, total);
+      }
   }
```

Comment:
We should have a VectorizedRecordReaderImpl that is preinitialized with the columns to read (columnSchemas).
It would also maintain the state.